frameworks/kafka/src/main/java/com/mesosphere/sdk/kafka/upgrade/KafkaConfigUpgrade.java

```diff
@@ -0,0 +1,369 @@
+                        .setMode(volume.getMode())
+                        .build();
+
+                Protos.Resource.DiskInfo.Builder newDiskInfoBuilder = diskInfo.toBuilder();
+                Protos.Resource.DiskInfo newDiskInfo = newDiskInfoBuilder.setVolume(newVolume)
+                        .setPersistence(diskInfo.getPersistence())
+                        .build();
+
+                Protos.Resource.Builder resourceBuilder = resource.toBuilder();
+                Protos.Resource newResource = resourceBuilder.setDisk(newDiskInfo)
+                        .setName(resource.getName())
+                        .setReservation(resource.getReservation())
+                        .setRole(resource.getRole())
+                        .setScalar(resource.getScalar())
+                        .build();
+
+                LOGGER.info("new disk resource : {}", newResource);
+
+                resourcesList.add(newResource);
+            }
+            taskInfoBuilder.clearResources();
+            taskInfoBuilder.addAllResources(resourcesList);
+
+            Protos.TaskInfo newTaskInfo = taskInfoBuilder.build();
+            LOGGER.info("Task {} new TaskInfo: ", newTaskInfo.getName(), newTaskInfo);
+            taskInfoList.add(newTaskInfo);
+        }
+        return taskInfoList;
+    }
+
+    private Map<String, Protos.TaskStatus> generateStatusMap(Collection<String> oldTaskNames)
+            throws KafkaConfigUpgradeException{
+        Map<String, Protos.TaskStatus> taskStatusMap = new HashMap<>();
+
+        for (String oldTaskName : oldTaskNames) {
+            int brokerId = oldTaskName2BrokerId(oldTaskName);
+            String newName = getNewTaskName(brokerId);
+            Optional<Protos.TaskStatus> optionalStatus = stateStore.fetchStatus(oldTaskName);
+            if (!optionalStatus.isPresent()){
+                throw new KafkaConfigUpgradeException("Can not fetch status for Task " + oldTaskName);
+            }
+            taskStatusMap.put(newName, optionalStatus.get());
+        }
+        return taskStatusMap;
+    }
+
+    private int oldTaskName2BrokerId(String taskName){
+        Pattern pattern = Pattern.compile("(.*)-(\\d+)");
+        Matcher matcher = pattern.matcher(taskName);
+        matcher.find();
+        return Integer.parseInt(matcher.group(2));
+    }
+
+    private String getNewTaskName(int brokerID){
+        return "kafka-" + brokerID + "-broker"; //kafka-2-broker
+    }
+
+    private Collection<String> getOldTaskNames(Collection<String> taskNames) {
+        Collection<String> oldNames = new ArrayList<>();
+        Pattern pattern = Pattern.compile("(.*)-(\\d+)");
+        for (String taskName: taskNames) {
+            Matcher matcher = pattern.matcher(taskName);
+            if (matcher.find()) {
+                oldNames.add(taskName);
+            }
+        }
+        return oldNames;
+    }
+
+    public static boolean enabled(){
+        if (System.getenv("CONFIG_UPGRADE_ENABLE") != null) {
```

Comment:
This is a standard way to handle booleans. There is no need to check the value.

If var is defined by Marathon, then it is true. If not defined by Marathon, then it is false. 

Please see marathon.mustache:L55, service.properties.mustache:L123, and DefaultScheduler:L364

Please note that false is not a boolean in shell. It is yet another string, or a predefined function, or a command /bin/false